IPython/core/interactiveshell.py

```diff
@@ -81,6 +81,18 @@
 from logging import error
 import IPython.core.hooks
 
+try:
+    import docrepr.sphinxify as sphx
+
+    def docformat(doc):
+        dirname = tempfile.mkdtemp(prefix='ipython_sphinxify_')
```

Comment:
For compatibility with Legacy Python, use our copy from `IPython.utils.tempdir.TemporaryDirectory`.